paradox-nakadi-consumer-boot/src/main/java/de/zalando/paradox/nakadi/consumer/boot/FailedEventReplayer.java

```diff
@@ -0,0 +1,88 @@
+package de.zalando.paradox.nakadi.consumer.boot;
+
+import static java.util.Objects.requireNonNull;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.util.Collection;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+import org.springframework.stereotype.Component;
+
+import de.zalando.paradox.nakadi.consumer.boot.components.EventReceiverRegistry;
+import de.zalando.paradox.nakadi.consumer.boot.components.EventTypeConsumer;
+import de.zalando.paradox.nakadi.consumer.boot.components.FailedEventSourceMap;
+import de.zalando.paradox.nakadi.consumer.core.EventHandler;
+import de.zalando.paradox.nakadi.consumer.core.FailedEventSource;
+import de.zalando.paradox.nakadi.consumer.core.domain.EventTypePartition;
+import de.zalando.paradox.nakadi.consumer.core.domain.FailedEvent;
+
+@Component
+public class FailedEventReplayer {
+
+    private final EventReceiverRegistry eventReceiverRegistry;
+
+    private final FailedEventSourceMap failedEventSourceMap;
+
+    private final ReplayHandler replayHandler;
```

Comment:
This is a public class but  `ReplayHandler`, which is passed through the constructor, is package local, i.e., this class is effectively package local.

Since `ReplayHandler` has no state, you can as well initialize this variable here, i.e., `private final ReplayHandler replayHandler = new ReplayHandler()`.